### PR TITLE
Display current FSM state in control panel

### LINKED ISSUE #50: [GUI] Display current state in control panel
Following from #26 and #49, we added socket handlers to send actions to the pod, but we'll also need to display the current state of the pod in the control panel. The socket handler in pod control will acknowledge the message which we can process in `PodSocketClient`. When the acknowledgement is received, we can update the `podData` state with a new property for the operating state of the pod.
- Create a new `State` enum in `PodSocketClient` to represent the operational state of the pod
- Add a new `state` property to the `PodData` interface
- For each of the methods to send the action message, in the acknowledgement callback, use `this.setPodData` to update the state of the pod
- In `ControlPanel`, display the `state` property of `podData`
  - It might be nice to show the full list of states and have a colored indicator next to the current state similar to an LED indicator

### PR DESCRIPTION
**Pull request by samderanova**
Resolves #50.

- Created a new enum `State` that represents the current state of the pod in the GUI frontend
- Modify `PodData` interface to include the current state as a field
- Display the current state in `ControlPanel`

### CODE REVIEW

File: control-station/src/services/PodSocketClient.ts [hunk 1]
```diff
@@ -64,24 +73,28 @@ class PodSocketClient {
 		this.socket.emit("load", (response: string) => {
 			console.log("Server acknowledged:", response);
 		});
+		this.setPodData((d) => ({ ...d, state: State.Load }));
```
Comment (taesungh) [inline control-station/src/services/PodSocketClient.ts:76]: Issue [repeating]: the control station must not falsely claim the pod state was updated when it actually might not have been. We'll need to update `podData` only when a proper acknowledgement is received (within the callback).

**Review by taesungh — CHANGES_REQUESTED** [review]
Thanks for working on this, one issue on usage and thoughts on initial value along with smaller chore.

**Review by taesungh — APPROVED** [review]
Looks alright for now, thanks for also adding the `Faulted` state. Might want to eventually consider using SASS or CSS modules for more organized styles.

**Review by ryescholin — APPROVED** [review]
Looks great!

**Pull request merged**